src/Entity/Subscription.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Entity;
+
+use App\Enum\SubscriptionFrequencyEnum;
+use App\Enum\SubscriptionSubjectEnum;
+use App\Repository\SubscriptionRepository;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity(repositoryClass: SubscriptionRepository::class)]
+class Subscription extends AbstractBaseEntity
+{
+    #[ORM\Column(length: 180, unique: false)]
+    private ?string $email = null;
+
+    #[ORM\Column(type: 'string', nullable: true, enumType: SubscriptionSubjectEnum::class)]
+    private ?SubscriptionSubjectEnum $subject = null;
+
+    #[ORM\Column(length: 180, unique: false)]
```

Comment:
use type: 'json' instead. Then you can avoid a lot of json_decode/encode

https://www.doctrine-project.org/projects/doctrine-dbal/en/4.1/reference/types.html#json

Then you get an array when retrieving from the db.
And you can feed the database with an array instead.